docs/source/install.rst

```diff
@@ -294,40 +274,167 @@ should run::
    whatever packages you just installed.
 
 
-Updating Firedrake
-------------------
+.. _customising:
 
-Updating Firedrake involves following the same steps as above when
-:ref:`installing Firedrake<install_firedrake>`. First, use ``firedrake-configure``
-to set the right environment variables and then run::
+Customising Firedrake
+=====================
 
-     $ pip install --upgrade git+https://github.com/firedrakeproject/firedrake.git
 
-Updating PETSc
-~~~~~~~~~~~~~~
+.. _firedrake_archs:
 
-To update PETSc you simply need to run::
+Prepared configurations
+-----------------------
 
-   $ cd petsc
-   $ git pull
-   $ make
+``firedrake-configure`` provides a number of different possible configurations
+(termed 'ARCHs') that specify how PETSc is configured and which external
+packages are built. The currently supported ARCHs are:
 
-This will only recompile PETSc's source code, not that of the external
-packages, and so should be relatively quick. If your PETSc is sufficiently
-out-of-date you may also need to rebuild the external packages by running::
+* ``default``: the default installation, suitable for most users
+* ``complex``: an installation where PETSc is configured using complex numbers
+
+The different configurations can be selected by passing the flag ``--arch`` to
+``firedrake-configure``. For example::
+
+   $ python3 firedrake-configure --show-petsc-configure-options --arch complex
+
+
+Optional dependencies
+---------------------
+
+PETSc with 64-bit indices
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To build PETSc with 64-bit indices you need to append ``--with-64-bit-indices``
+to the set of options when PETSc is configured. For example this could be::
+
+   $ python3 ../firedrake-configure --show-petsc-configure-options | xargs -L1 ./configure --with-64-bit-indices
+
```

Comment:
```suggestion
Some of the packages that PETSc uses must be configured specially for 64-bit indices. If PETSc downloads a package (i.e. `--download-package` is in the PETSc configure options) then it will ensure that the package is configure correctly.
However, if a package has been installed externally (i.e. `--with-package` or `--with-package-{lib,include}=...` are in the PETSc configure options), then it is your responsibility to either a) ensure that it is configured correctly for 64-bit integers, or b) replace the PETSc configure option with `--download-package` so that PETSc handles the configuration.

```

Reply:
I've included something very similar to this